### PR TITLE
reverse sequence

### PR DESCRIPTION
**Pull request by lucienwang1009**
support MS ReverseSequence.

### CODE REVIEW

File: tests/test_backend.py [hunk 1]
```diff
@@ -1572,15 +1576,35 @@ def test_reverse_sequence_time_major(self):
...
+    def test_reverse_sequence_batch_major(self):
+        self._test_reverse_sequence_batch_major()
+
+    @check_opset_min_version(8, "Scan")
+    @skip_opset(9, "ReverseSequence")
+    def test_reverse_sequence_time_major(self):
+        self._test_reverse_sequence_time_major()
+
+    @test_ms_domain()
+    @unittest.skipIf(True, "not support in pypi onnxruntime")
```
Comment (nbcsm) [inline tests/test_backend.py]: skip if onnxruntime version < 0.4?
so we don't need to enable this test explicitly for next onnxruntime release.

File: tests/common.py [hunk 2]
```diff
@@ -177,6 +177,21 @@ def skip_onnxruntime_backend(message=""):
     return unittest.skipIf(config.is_onnxruntime_backend, reason)
 
 
+def check_onnxruntime_backend(message=""):
+    """ Skip if backend is onnxruntime """
```
[the remark this answers is not in the document]
Comment (nbcsm) [reply tests/common.py]: [optional] is that possible to apply ```check_onnxruntime_backend``` automatically within ```test_ms_domain```?

File: tests/common.py [hunk 3]
```diff
@@ -177,6 +177,21 @@ def skip_onnxruntime_backend(message=""):
...
+    reason = _append_message("only supported by onnxruntime", message)
+    return unittest.skipIf(not config.is_onnxruntime_backend, reason)
+
+
+def check_onnxruntime_min_version(min_required_version, message=""):
+    """ Skip if onnxruntime version < min_required_version """
+    config = get_test_config()
+    reason = _append_message("conversion requires onnxruntime >= {}".format(min_required_version), message)
+    return unittest.skipIf(config.is_onnxruntime_backend and
+                           config.backend_version < LooseVersion(min_required_version), reason)
```
Comment (nbcsm) [inline tests/common.py]: skip if backend is NOT onnxruntime OR onnxruntime version < min_required_version?
so no need for check_onnxruntime_backend too.

File: tests/test_backend.py [hunk 4]
```diff
@@ -1572,15 +1576,38 @@ def test_reverse_sequence_time_major(self):
...
+    @check_opset_min_version(8, "Scan")
+    @skip_opset(9, "ReverseSequence")
+    def test_reverse_sequence_time_major(self):
+        self._test_reverse_sequence_time_major()
+
+    # only support onnxruntime with version larger than 0.4.0
+    @test_ms_domain()
+    @check_onnxruntime_min_version("0.4.0")
+    @check_onnxruntime_backend()
+    def test_ms_reverse_sequence_batch_major(self, extra_opset):
```
[the remark this answers is not in the document]
Comment (lucienwang1009) [reply tests/test_backend.py]: Yes, it's passing.

File: tests/common.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucienwang1009) [reply tests/common.py]: np, will add it to `test_ms_domain`

File: tests/common.py (hunk 3 above)
Comment (nbcsm) [reply tests/common.py]: after discussion, decided to keep current semantics, similar to ```check_onnxruntime_incompatibility```

**Pull request merged**